### PR TITLE
Implement lottery logic

### PR DESCRIPTION
**Pull request by dSkarzh**

### CODE REVIEW

File: src/main/java/core/basesyntax/Ball.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Ball {
+    private final Color color;
+    private final int number;
+
+    protected Ball(Color color, int number) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Ball.java:7]: why protected? the usage of protected is very limited in practice, let's better be not afraid to use public

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
+
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    protected Random random = new Random();
+
+    protected Color getRandomColor() {
+        int index = random.nextInt(Color.values().length);
+        return Color.values()[index];
     }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/ColorSupplier.java:11]: same here about protected

File: src/main/java/core/basesyntax/Lottery.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import java.util.Random;
+
+public class Lottery {
+    private static final int BOUND = 100;
+    protected ColorSupplier colorSupplier = new ColorSupplier();
+    protected Random random = new Random();
+
+    public Ball getRandomBall() {
+        return new Ball(colorSupplier.getRandomColor(), random.nextInt(BOUND));
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Lottery.java:12]: great job on constants and working with fields in service classes

if you want to go 1 step further, you can pass `new ColorSupplier()` and `new Random()` objects into the Lottery's constructor from `main()`

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Ball.java (hunk 1 above)
Comment (dSkarzh) [reply src/main/java/core/basesyntax/Ball.java:7]: are you talking about this place specificly or about access modifiers in whole project?

File: src/main/java/core/basesyntax/Lottery.java (hunk 3 above)
Comment (dSkarzh) [reply src/main/java/core/basesyntax/Lottery.java:17]: Thanks! Just commited proposed changes

File: src/main/java/core/basesyntax/Ball.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/core/basesyntax/Ball.java]: in whole project

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**